Kin Unity/Assets/Kin/Editor/NativeCode/iOS/KinPlugin.swift

```diff
@@ -24,6 +25,23 @@ struct Provider: ServiceProvider {
 	}
 }
 
+enum backupAction {
+    case BACKUP
+    case RESTORE
+}
+
+// Struct to cache the params the were requested to use in the callbacks
+struct BackupRestoreRequest {
```

Comment:
Because you can only have either a backup with account id or restore with client id (and no other combinations) you should change the struct to an [associated enum](https://docs.swift.org/swift-book/LanguageGuide/Enumerations.html#ID148). Also enums, like structs and classes follow the pascal case (upper camel case) pattern while their properties follow the camel case pattern.